src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { jwtConstants } from './constants';
+import { UserDummy } from '../users/user-dummy';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor() {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+      ignoreExpiration: false,
+      secretOrKey: jwtConstants.secret,
+    });
+  }
+
+  async validate(payload: any): Promise<UserDummy> {
+    return { id: payload.sub, username: payload.username } as UserDummy;
```

Comment:
Genuine question - since you're defining `password` as a property on the `UserDummy` type, wouldn't this throw a type issue here? Or is this avoided by casting it explicity by `as UserDummy`?